tests/api/notifications/get.test.js

```diff
@@ -5,7 +5,8 @@ const { authReq } = __.require('apiTests', 'utils/utils')
 
 describe('notifications:get', () => {
   it('should get user notifications', async () => {
-    const { notifications } = await authReq('get', '/api/notifications')
+    const { notifications, total } = await authReq('get', '/api/notifications')
```

Comment:
sooo, lets say this part of the code is far from having a good test coverage :) ... Since some commits are done, it would be great to detail a bit the tests to check what is a notification (ie. its type). But I dont think its necessary to go as far as testing the pagination itself.

a notification is triggered when making new friends: 

```js
const { makeFriends } = __.require('apiTests', 'utils/relations')
...
await makeFriends(user, friend)
```

again same comment as above this suggestion is optional

Reply:
I kept the `total` assertion to ensure the pagination change has taken effect and added a notification as you said but I have seemingly random failure on this test (like one failure after running the test 30 times)

I got "lucky" the first time I guess, then started loop-testing the function.

Is it possible that the `makeFriends`' notification is not directly available for reading? Should I add a `sleep(1)`?

Reply:
i cant reproduce the failure.
Indeed you may want to introduce a "sleep" with the `wait` function in `lib/promises`, even though i dont see why would be necessary.

Reply:
short version: adding a 1s delay should work in most cases (that is, when the server isn't under heavy fire?). Or wait for #446 to be merged.

longer version: when accepting the friend request, a `notify:friend:request:accepted` event is triggered, letting the [notifications event listener trigger the `acceptedRequest` callback](https://github.com/inventaire/inventaire/blob/874b095/server/lib/notifications.js#L112), but the server doesn't wait for that async callback to be done to answer, so when waiting for `await makeFriends`, you have no guaranty that the notification was already posted on the notifications database... 

Those event listeners are not very developer-friendly for tests, but they express the idea that notifications/emails/tracking events are not essential: if the server fails to do one of those, that shouldn't make the HTTP request return an error. But, in tests, having that guaranty would be preferable, so here is a fix proposition https://github.com/inventaire/inventaire/pull/446

Reply:
btw, to reliably reproduce the failure (and thus test the #446 patch), you can turn that `acceptedRequest` function into an `async` function and add `await wait(5000)`

Reply:
I was suspecting something like that, and I'll wait for the patch then. Thanks for the explanation !

Reply:
I just merged #446 so rebasing on master should solve your issue

Reply:
done